### PR TITLE
add doc for arctern_spark and try to be consistent with pandas arctern

### PR DESCRIPTION
**Pull request by shengjh**
Please help to review the doc. @Emma-Song

### CODE REVIEW

File: spark/arctern_spark/geodataframe.py [hunk 1]
```diff
@@ -157,6 +201,46 @@ def copy(self, deep=None):
         return gdf
 
     def dissolve(self, by, col="geometry", aggfunc="first", as_index=True):
+        """
+        Dissolves geometries within `by` into a single observation.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: use double quotes as code format, for example \`\`by\`\`

File: spark/arctern_spark/geodataframe.py [hunk 2]
```diff
@@ -157,6 +201,46 @@ def copy(self, deep=None):
...
+
+        This is accomplished by applying the `unary_union` method to all geometries within a group.
+
+        Observations associated with each `by` group will be aggregated using the `aggfunc`.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: \`\`by\`\` and \`\`aggfunc\`\`

File: spark/arctern_spark/geodataframe.py [hunk 3]
```diff
@@ -157,6 +201,46 @@ def copy(self, deep=None):
...
+
+        Parameters
+        ----------
+        by: str
+            Column whose values define groups to be dissolved, by default None.
+        aggfunc: function or str
+            Aggregation function for manipulation of data associated with each group, by default "first". Passed to pandas `groupby.agg` method.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: \`\`groupby.agg\`\`

**Review by Emma-Song — CHANGES_REQUESTED** [review]

**Review by Emma-Song — APPROVED** [review]

**Comment by shengjh** [comment]
Additionally, `GeoDataFrame::init `, `sjoin`, `GeoDataFrame::merge` doc have been updated. Please leave a review. Thanks
@Emma-Song

File: spark/arctern_spark/geodataframe.py [hunk 4]
```diff
@@ -30,6 +30,55 @@
 
 class GeoDataFrame(DataFrame):
     def __init__(self, data=None, index=None, columns=None, dtype=None, copy=False, geometries=None, crs=None):
+        """
+        A GeoDataFrame object is a pandas.DataFrame that has columns with geometry.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: A GeoDataFrame object is a pandas.DataFrame that has columns with **geometries.**

File: spark/arctern_spark/geodataframe.py [hunk 5]
```diff
@@ -30,6 +30,55 @@
...
+
+        Parameters
+        ----------
+        data : numpy ndarray (structured or homogeneous), dict, pandas DataFrame, Spark DataFrame \
+            Koalas Series, or GeoSeries
+            Dict can contain Series, GeoSeries, arrays, constants, or list-like objects
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: add a full stop `.` at the end

File: spark/arctern_spark/geodataframe.py [hunk 6]
```diff
@@ -30,6 +30,55 @@
...
+            If data is a dict, argument order is maintained for Python 3.6
+            and later.
+            Note that if `data` is a pandas DataFrame, a Spark DataFrame, a Koalas Series, and a GeoSeries,
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: use double quote for code format

File: spark/arctern_spark/geodataframe.py [hunk 7]
```diff
@@ -30,6 +30,55 @@
...
+            other arguments should not be used.
+        index : Index or array-like
+            Index to use for resulting frame. Will default to RangeIndex if
+            no indexing information part of input data and no index provided.
+        columns : Index or array-like
+            Column labels to use for resulting frame. Will default to
+            RangeIndex (0, 1, 2, ..., n) if no column labels are provided.
+        dtype : dtype, default None
+            Data type to force. Only a single dtype is allowed. If None, infer.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: If None, infers the data type based on the data.

File: spark/arctern_spark/geodataframe.py [hunk 8]
```diff
@@ -30,6 +30,55 @@
...
+        copy : bool, default False
+            Copy data from inputs. Only affects DataFrame / 2d ndarray input.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: Copys
DataFrame **or** 2d ndarray input

File: spark/arctern_spark/geodataframe.py [hunk 9]
```diff
@@ -30,6 +30,55 @@
...
+        geometries : list
+            The name of columns which are setten as geometry columns.
+        crs : str or list, default None
+            Coordinate Reference System of the geometry objects.
+            if crs is a string, will set all ``geometries`` columns' crs to param ``crs``.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: if > If
crs > \`\`crs\`\`
will set > sets

File: spark/arctern_spark/geodataframe.py [hunk 10]
```diff
@@ -71,7 +120,10 @@ def _set_geometries(self, cols, crs=None):
 
     def set_geometry(self, col, inplace=False, crs=None):
         """
-        Sets an existing column in the GeoDataFrame to a geometry column, which is used to perform geometric calculations later.
+        Set an existing column in the GeoDataFrame to a geometry column, which is used to perform geometric calculations later.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: Sets

File: spark/arctern_spark/geodataframe.py [hunk 11]
```diff
@@ -71,7 +120,10 @@ def _set_geometries(self, cols, crs=None):
...
+
+        Setting an column to a geometry column will attemp to construct geometry for each row of this column,
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py:125]: This function attempts to construct geometries for each row of the given column, so all data in \`\`col\`\` should be in WKT or WKB format.

File: spark/arctern_spark/geodataframe.py [hunk 12]
```diff
@@ -262,12 +314,51 @@ def merge(
             suffixes=("_x", "_y")
     ):
         """
-        Merges two GeoDataFrame objects with a database-style join.
+        Merge GeoDataFrame objects with a database-style join.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py:317]: Merges

File: spark/arctern_spark/geodataframe.py [hunk 13]
```diff
@@ -262,12 +314,51 @@ def merge(
...
+
+        The index of the resulting GeoDataFrame will be one of the following:
+            - 0...n if no index is used for merging
+            - Index of the left GeoDataFrame if merged only on the index of the right GeoDataFrame
+            - Index of the right GeoDataFrame if merged only on the index of the left GeoDataFrame
+            - All involved indices if merged using the indices of both GeoDataFrames
+                e.g. if `left` with indices (a, x) and `right` with indices (b, x), the result will
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py:324]: code format

File: spark/arctern_spark/geodataframe.py [hunk 14]
```diff
@@ -262,12 +314,51 @@ def merge(
...
+
+            * *left:* use only keys from left frame, similar to a SQL left outer join; preserve key
+                order.
+            * *right:* use only keys from right frame, similar to a SQL right outer join; preserve key
+                order.
+            * *outer:* use union of keys from both frames, similar to a SQL full outer join; sort keys
+                lexicographically.
+            * *inner:* use intersection of keys from both frames, similar to a SQL inner join;
+                preserve the order of the left keys.
+        on: Column or index level names to join on. These must be found in both GeoDataFrames. If on
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: If \`\`on\`\` is None

File: spark/arctern_spark/geodataframe.py [hunk 15]
```diff
@@ -262,12 +314,51 @@ def merge(
...
+            is None and not merging on indexes then this defaults to the intersection of the
+            columns in both GeoDataFrames.
+        left_on: Column or index level names to join on in the left GeoDataFrame. Can also
+            be an array or list of arrays of the length of the left GeoDataFrame.
+            These arrays are treated as if they are columns.
+        right_on: Column or index level names to join on in the right GeoDataFrame. Can also
+            be an array or list of arrays of the length of the right GeoDataFrame.
+            These arrays are treated as if they are columns.
+        left_index: Use the index from the left GeoDataFrame as the join key(s). If it is a
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: Uses

File: spark/arctern_spark/geodataframe.py [hunk 16]
```diff
@@ -262,12 +314,51 @@ def merge(
...
+            MultiIndex, the number of keys in the other GeoDataFrame (either the index or a number of
+            columns) must match the number of levels.
+        right_index: Use the index from the right GeoDataFrame as the join key. Same caveats as
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: Uses

File: spark/arctern_spark/geodataframe.py [hunk 17]
```diff
@@ -262,12 +314,51 @@ def merge(
...
+            left_index.
+        suffixes: Suffix to apply to overlapping column names in the left and right side,
+            respectively.
 
         Returns
         -------
-            GeoDataFrame or pandas.DataFrame
-            Returns a GeoDataFrame if a geometry column is present; otherwise, returns a pandas DataFrame.
+            GeoDataFrame
+            Return a merged GeoDataFrame.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: Returns

File: spark/arctern_spark/geodataframe.py [hunk 18]
```diff
@@ -388,7 +479,7 @@ def reset_index(self, level=None, drop=False, inplace=False, col_level=0, col_fi
     @classmethod
     def from_file(cls, filename, **kwargs):
         """
-        Constructs a GeoDataFrame from a file or url.
+        Construct a GeoDataFrame from a file or url.
```
Comment (Emma-Song) [inline spark/arctern_spark/geodataframe.py]: The subject is "the function", so should add a 's' to all verbs. Please change the words below.

File: spark/arctern_spark/sjoin.py [hunk 19]
```diff
@@ -30,12 +30,10 @@ def sjoin(
         Specify geometry columns of left_df, right_df to be joined.
     how : string, default 'inner'
         The type of join:
-
-        # TODO: fix doc
-        * 'left': use keys from left_df; retain only left_df geometry column
-        * 'right': use keys from right_df; retain only right_df geometry column
-        * 'inner':
-        * 'full':
+        * *left:* use keys from left_df
```
Comment (Emma-Song) [inline spark/arctern_spark/sjoin.py]: Uses

**Review by Emma-Song — CHANGES_REQUESTED** [review]

**Comment by shengjh** [comment]
All problems have been solved(i hope), please review again @Emma-Song

**Review by Emma-Song — APPROVED** [review]

**Pull request merged**